lib/Doctrine/DBAL/Schema/MySqlSchemaManager.php

```diff
@@ -176,17 +177,26 @@ protected function _getPortableTableColumnDefinition($tableColumn)
                 break;
         }
 
-        $length = ((int) $length == 0) ? null : (int) $length;
+        $length = ((int) $length) ?: null;
+
+
+        $isNotNull = ($tableColumn['null'] !== 'YES');
+
+        if ($this->_platform instanceof MariaDb1027Platform) {
+            $columnDefault = $this->getMariaDb1027ColumnDefault($tableColumn['default']);
```

Comment:
It looks inconsistent that this branch of the `if` statement doesn't check whether the default value is set, but the `else` branch does.